.github/workflows/main.yml

```diff
@@ -0,0 +1,94 @@
+name: Godot CI
+
+on:
+    pull_request:
+    push:
+      branches: [ master, 3.2, 3.1, 3.0 ]
+env:
+  SCONS_CACHE_MSVC_CONFIG: true
+  SCONS_CACHE_LIMIT: 2048
+jobs:
+  windows-editor:
+    runs-on: "windows-latest"
+    name: Windows Editor (release_debug)
+    steps:
+    - uses: actions/checkout@v2 
+    # cache must be put in the github workspace directory on windows to be persisted correctly
+    - name: "force cache path to be correct"
+      run: echo "::set-env name=SCONS_CACHE::${{env.HOME}}/.scons_cache/"
+    # upload cache on completion and check it out now
+    - name: configure .scons_cache and download cache
+      id: scons-cache
+      uses: RevoluPowered/cache@v2.1
+      with:
+        path: ${{env.SCONS_CACHE}}
+        key: ${{ runner.os }}-godot-release_debug_editor
+        restore-keys: ${{ runner.os }}-godot-release_debug_editor
+    # use python 3.x release (works cross platform)
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v2
+      with:
+        # Semantic version range syntax or exact version of a Python version
+        python-version: '3.x' 
+        # Optional - x64 or x86 architecture, defaults to x64
+        architecture: 'x64' 
+    # You can test your matrix by printing the current Python version
+    - name: Display Python version
+      run: python -c "import sys; print(sys.version)"
+    - name: Installing scons and pywin32
+      run: |
+        python -m pip install scons pywin32
+    - name: Check versioning
+      run: |
+        python --version
+        scons --version
+    - name: Compilation
+      run: |
+        scons -j2 verbose=yes warnings=all werror=yes platform=windows target=release_debug
```

Comment:
Is it confirmed that GitHub Actions has 2 cores, and only 2?

Reply:
Yes only 2 cores per machine, you only get more if you pay, or if you self host.

side topic: Self hosting is fairly easy and has been tested already it works, with very minor tweaks. Requires a dedicated vm which is secured to work for the godot engine though, so would recommend we stay with their stuff until @hpvb can look at setting something up possibly?

I tested with windows 10 already and the self hosting feature just works with no specific or non sane options, one minor edit to the permissions on the vm and it works. To be clear this config explictly will only build using the Github Action servers right now.